os/arch/arm/src/common/up_mpuinit.c

```diff
@@ -50,11 +50,119 @@
 /****************************************************************************
  * Pre-processor Definitions
  ****************************************************************************/
+static struct mpu_usages_s g_mpu_region_usages;
+
+#ifdef CONFIG_ARMV7M_MPU
+#define MPU_MAX_NREGION		CONFIG_ARMV7M_MPU_NREGIONS
+#elif defined(CONFIG_ARMV8M_MPU)
+#define MPU_MAX_NREGION		CONFIG_ARMV8M_MPU_NREGIONS
+#endif
 
 /****************************************************************************
  * Public Functions
  ****************************************************************************/
 
+/****************************************************************************
+ * Name: mpu_regs_initialize
+ *
+ * Description:
+ *   Initialize the MPU region numbers.
+ *
+ ****************************************************************************/
+
+void mpu_regs_initialize(FAR struct mpu_usages_s *mpu)
+{
+
+	uint8_t offset = mpu->nregion_board_specific;
+
+#ifdef CONFIG_SUPPORT_COMMON_BINARY
+#ifdef CONFIG_OPTIMIZE_APP_RELOAD_TIME
+	offset += 3;
```

Comment:
Could you add comment why you add 3 or 1 in case of each config?

Reply:
This has been supported while configuring MPU regions. So i think not needed here again.

Reply:
I found the `MPU_NUM_REGIONS` in <tinyara/mpu.h>. It looks the same purpose of this.
Could you check that below is right?
```c
#ifdef CONFIG_SUPPORT_COMMON_BINARY
     offset += MPU_NUM_REGIONS;
     mpu->nregion_common_bin = offset;
#endif 

#ifdef CONFIG_APP_BINARY_SEPARATION
     offset += MPU_NUM_REGIONS;
     mpu->nregion_app_bin = offset;
#endif 
```